spec/controllers/activity_deletions_controller_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "rails_helper"
+
+RSpec.describe ActivityDeletionsController do
```

Comment:
The commit message around this confused me a bit, I think it should say something like this instead.

```
The policy dependency of the controller is mocked so that we are only testing 
the controller itself.
```

Rather than:

```
For the controller spec, I wanted to follow the single responsibility
principle, using contexts of "when authorised" and "when unauthorised",
rather than setting up the various combinations of activity level, user
organisation and titled-ness, which should be the concern of the policy.
It took a while to work out how to make the controller not care about
the user, so I considered reverting to a complex context structure, but
in the end I worked out a way
```

For two reasons:
1. I think the details about the journey of how it was implemented are irrelevant unless they may be important in future (sadly!)
2. I read it as the `single responsibility` of the spec itself, which doesn't make sense. Whereas, I think its saying "For the controller spec, I wanted to test only the responsibility of the class (controller)"


Hope that made sense!

Reply:
I think _technically_ single responsibility only relates the the responsibility of a class, which probably led to my confusion. Though I've definitely heard it being used to describe stuff like APIs 

I'm not sure if this is picky? If it's leading to confusion, then maybe not?

Reply:
😮 interesting. I thought it was just a general principle in all code (including test code) to be followed to varying degrees. In any case, I think I can clarify the wording here!

Reply:
Don't take my word for it!

[This](https://stackify.com/solid-design-principles/) article starts by defining it as a class thing, then goes on to mention microservices. Maybe it's original meaning was in relation to classes, not it's a more wooly term 🤷

Reply:
Nice! What book is this?

Reply:
https://www.google.co.uk/books/edition/Spring_5_0_Microservices/CedDDwAAQBAJ?hl=en&gbpv=1&dq=single+responsibility+principle+%22microservices%22&pg=PA128&printsec=frontcover 

I just went to books as a trustable resource. 

If we see Uncle Bob as the ultimate truth on the matter, then I think that second quote he is elaborating on the first.
Found [this tweet](https://twitter.com/unclebobmartin/status/1544655247305490432?s=20) that suggests that too

Reply:
I've reworded this, incorporating your suggestion while making the detail more of an explanation of the implementation instead of recounting the journey 😅